### DISCUSSION TITLE
Coinbase Wallet Slow

### DISCUSSION THREAD

**Discussion by a-znk** [wevm/wagmi#870] [Q&A]
Hey,

I've run into a strange issue... when connecting to my dapp using Coinbase Wallet, the whole application freezes up and gets very choppy, often becoming unresponsive. So far, I've tried switching from the public provider to Alchemy, testing on desktop and mobile devices, but no luck pinpointing the issue... I also believe i have narrowed the issue down to the minting logic in my dapp... It could be right under my nose, but I'm wondering if anyone has experienced this issue before? Final note - this is currently happening in development and production, and I am using Goerli network at the moment.

Thank you!

**Comment by bigint** [comment]
I also face this on https://lenster.xyz 😕

**Comment by tmm** [comment]
Using the browser extension or mobile app?

**Reply by bigint** [reply]
It happens for me on both 🙇🏼

**Reply by a-znk** [reply]
Same for me.

**Reply by tmm** [reply]
Minimal reproducible example would be great.

My guess is it's related to a [Coinbase Wallet release](https://github.com/coinbase/coinbase-wallet-sdk/releases), but I'm not able to reproduce. You could play around with the installed version and see if that fixes it.

**Reply by a-znk** [reply]
Thanks for your reply. That could certainly make sense about a recent release. Here is the code that is causing the issue for me, if that helps... Simply removing this bit of code "fixes" the issue. No more freezing. However, the dapp I'm building is for minting nfts, so that isn't a good solution of course. I've tried several different iterations for this minting code, but the issue persists with all. To reiterate, this problem is exclusive to Coinbase. It functions perfectly with other wallets.

`  const { config } = usePrepareContractWrite({
    ...contractConfig,
    functionName: "mint",
    args: [mintAmount],
    overrides: {
      value: costData?.mul(mintAmount),
    },
  });

  const { write: mint } = useContractWrite(config);`

**Reply by sjshon** [reply]
@tmm tried with coinbase versions 3.4.2, 3.4.1, 3.4.0, 3.3.0, but issue persisted.
@a-znk you are right, i removed both `usePrepareContractWrite` and `useContractWrite` then coinbase started not freezing up the site @tmm are you be able to reproduce the problem with those two hooks in place?

**Comment by a7ek** [comment]
same here

**Comment by sjshon** [comment]
experiencing the same issue, whole site freezes up

**Comment by sjshon** [comment]
@tmm alright so seems like no one's trying to solve when this is a legit problem and end-users started to argue. I ended up creating a [minimal reproducible example.](https://github.com/kkagill/wagmi-coinbase-freeze) Very simple, instructions are written on readme. Please take a look :)

**Reply by taycaldwell** [reply]
Seems specific to `usePrepareContractWrite`. I'm seeing no issue with other prepare hooks like `usePrepareSendTransaction`, or when I modify the function called by the button.

**Reply by sjshon** [reply]
yeah i ended up removing it and replaced that with web3.js's `estimateGas`

**Comment by taycaldwell** [comment]
@tmm This seems to be effecting a lot of Coinbase Wallet users, and I'm trying to determine if it's an issue with wagmi or Coinbase Wallet SDK.

Can you take a look to make sure it's not an issue on the wagmi side?  @sjshon provided a sample above.

**Reply by tmm** [reply]
Was on vacation when I initially responded and just got back. I was able to reproduce with the example and will take a more detailed look to see what's going on.

**Reply by sjshon** [reply]
@tmm has this been resolved?

**Reply by ReachCambo** [reply]
Hi
